client/src/app/main/machine/upload/machine-upload.component.ts

```
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { MachineService } from '../../../shared/services/machine/machine.service';
+import { FormGroup, FormControl } from '@angular/forms';
+import { Machine } from '../../../shared/interfaces/machine.interface';
+
+@Component({
+  templateUrl: './machine-upload.component.html',
+  styleUrls: ['./machine-upload.component.scss']
+})
+export class MachineUploadComponent implements OnInit {
+  public file;
+  public uploadCsvForm: FormGroup;
+  public machines: Machine[];
+
+  constructor(public machineService: MachineService) {
+    this.uploadCsvForm = new FormGroup({
+      firstName: new FormControl()
+    });
+  }
+
+  public ngOnInit() {
+    this.machineService.find()
+      .subscribe((machines: Machine[]) => {
+        this.machines = machines;
+      }, (error: any) => {
+        throw new Error(error);
+      });
+  }
+
+  public onFileChange(event): void {
+    this.file = event.target.files[0];
+  }
+
+  public uploadCsv(): void {
+    this.machineService.uploadCsv(this.file).subscribe(
+      value => console.log(value),
+      err => console.log(err)
```

you should remove console logs. you can throw the error, we will need a message service to give a success message for the success in the future